partiql-spi/src/main/java/org/partiql/spi/value/DatumComparator.java

```diff
@@ -558,24 +558,12 @@ private static DatumComparison[] fillTimeComparator(DatumComparison[] comps) {
         comps[TIMEZ] = (self, time, comp) -> self.getOffsetTime().compareTo(time.getOffsetTime());
         return comps;
     }
-    
-    private static final long SECONDS_PER_MINUTE = 60;
-    private static final long MINUTES_PER_HOUR = 60;
-    private static final long HOURS_PER_DAY = 24;
-    private static final long SECONDS_PER_HOUR = SECONDS_PER_MINUTE * MINUTES_PER_HOUR;
-    private static final long SECONDS_PER_DAY = SECONDS_PER_HOUR * HOURS_PER_DAY;
 
     @SuppressWarnings({"UnusedReturnValue"})
     private static DatumComparison[] fillIntervalComparator(DatumComparison[] comps) {
-        comps[INTERVAL_YM] = (self, other, comp) -> {
-            long totalMonths = self.getYears() *  (long) 12 + self.getMonths();
-            long otherTotalMonths = other.getYears() * (long) 12 + other.getMonths();
-            return Long.compare(totalMonths, otherTotalMonths);
-        };
+        comps[INTERVAL_YM] = (self, other, comp) -> Long.compare(self.getTotalMonths(), self.getTotalMonths());
         comps[INTERVAL_DT] = (self, other, comp) -> {
-            long totalSeconds = self.getDays() * SECONDS_PER_DAY + self.getHours() * SECONDS_PER_HOUR + self.getMinutes() * SECONDS_PER_MINUTE + self.getSeconds();
-            long otherTotalSeconds = other.getDays() * SECONDS_PER_DAY + other.getHours() * SECONDS_PER_HOUR + other.getMinutes() * SECONDS_PER_MINUTE + other.getSeconds();
-            int comparison = Long.compare(totalSeconds, otherTotalSeconds);
+            int comparison = Long.compare(self.getTotalSeconds(), other.getTotalSeconds());
```

Comment:
self-review: this was another place we can simplify the computation between intervals